src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java

```diff
@@ -202,6 +202,9 @@ private Configuration getBaseConfiguration(@Nullable String cmekKey) throws IOEx
                                  getConfig().isAllowSchemaRelaxation());
     baseConfiguration.setStrings(BigQueryConfiguration.OUTPUT_TABLE_WRITE_DISPOSITION_KEY,
                                  getConfig().getWriteDisposition().name());
+    // this setting is needed because gcs has default chunk size of 64MB. This is large default chunk size which can
+    // cause OOM issue if there are many tables being written. See this - CDAP-16670
+    baseConfiguration.set("fs.gs.outputstream.upload.chunk.size", "8388608");
```

Comment:
Can we make this configurable?

Reply:
If we make it configurable then we to have best practice guide for that please.

Reply:
The default should be 8M, but making it configuration to give more flexibility for wider use cases